File: 2018.08.17 Speed Comparison of C vs PyPy2.7 vs Swift 4/Attachments/sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void sieve_eratosthenes(unsigned int n)
{
	int *sieve = calloc(n, sizeof(unsigned int));
	sieve[0] = 1;
	for (int p = 2; p <= n; ++p) {
		if (sieve[p - 1] == 0) {
			/* printf("%d\n", p); */
			for (int i = p * p; i <= n; i += p)
				sieve[i - 1] = 1;
		}
	}
}

int main(void)
{
	unsigned int n = 100000000;
	double average = 0.0;
	for (int i = 0; i < 100; i++) {
		clock_t start = clock();
		sieve_eratosthenes(n);
		clock_t end = clock();
		average += (double) (end - start) / CLOCKS_PER_SEC;
	}
	printf("%f\n", average / 100.);
	return 0;
}

File: 2018.08.17 Speed Comparison of C vs PyPy2.7 vs Swift 4/Attachments/sieve.py
import time


def sieve_eratosthenes(n):
    sieve = [1] * n
    sieve[0] = 0
    for p in xrange(2, n + 1):
        if sieve[p - 1]:
            # print p
            for i in xrange(p ** 2, n + 1, p):
                sieve[i - 1] = 0


n = 100000000
average = 0
for _ in xrange(2):
    start = time.time()
    sieve_eratosthenes(n)
    end = time.time()
    average += end - start
print average / 2.
